Framework/Core/include/Framework/InputRecord.h

```diff
@@ -29,11 +29,74 @@
 #include <memory>
 #include <type_traits>
 
+template <typename T>
+using default_delete = std::default_delete<T>;
+
 namespace o2 {
 namespace framework {
 
 struct InputSpec;
 
+/// A deleter type to be used with unique_ptr, which can be marked that
+/// it does not own the underlying resource and thus should not delete it.
+/// The resource ownership property controls the behavior and can only be
+/// set at construction of the deleter in the unique_ptr. Falls back to
+/// default_delete if not initialized to 'NotOwning'.
+/// Usage: unique_ptr<T, Deleter<T>> ptr(..., Deleter<T>(false))
+///
+/// By contract, the underlying, not owned resource is supposed to be
+/// available during the lifetime of the object, which is the case in the
+/// InputRecord and DPL processing APIs. The Deleter can be extended
+/// to support a callback to call a resource management outside.
+template <typename T>
+class Deleter : public default_delete<T>
+{
+ public:
+  enum class OwnershipProperty : short {
+    Unknown = -1,
```

Comment:
I think our enums should be all capital. Also I would use `enum struct` rather than `enum class`.

Reply:
ok

Reply:
will change to `enum struct`, as for the names the coding conventions say camel case convention, so they should be fine